osu.Game/Database/LegacyImportManager.cs

```diff
@@ -54,17 +54,37 @@ public partial class LegacyImportManager : Component
 
         public void UpdateStorage(string stablePath) => cachedStorage = new StableStorage(stablePath, gameHost as DesktopGameHost);
 
-        public bool CheckHardLinkAvailability()
+        /// <summary>
+        /// Check whether importing some content from stable can make use of hard links
+        /// </summary>
+        /// <param name="content">Content to check for</param>
+        /// <returns>Whether all content checked can use hard links</returns>
+        public bool CheckHardLinkAvailability(StableContent content = StableContent.All)
         {
             var stableStorage = GetCurrentStableStorage();
 
             if (stableStorage == null || gameHost is not DesktopGameHost desktopGameHost)
                 return false;
 
-            string testExistingPath = stableStorage.GetFullPath(string.Empty);
-            string testDestinationPath = desktopGameHost.Storage.GetFullPath(string.Empty);
+            bool allCanUse = true;
 
-            return HardLinkHelper.CheckAvailability(testDestinationPath, testExistingPath);
+            if (content.HasFlagFast(StableContent.Beatmaps))
+            {
+                string testExistingPath = stableStorage.GetSongStorage().GetFullPath(string.Empty);
+                string testDestinationPath = desktopGameHost.Storage.GetFullPath(string.Empty);
+
+                allCanUse = allCanUse && HardLinkHelper.CheckAvailability(testDestinationPath, testExistingPath);
+            }
+
+            if (content.HasFlagFast(StableContent.Scores) || content.HasFlagFast(StableContent.Skins) || content.HasFlagFast(StableContent.Collections))
+            {
+                string testExistingPath = stableStorage.GetFullPath(string.Empty);
+                string testDestinationPath = desktopGameHost.Storage.GetFullPath(string.Empty);
+
+                allCanUse = allCanUse && HardLinkHelper.CheckAvailability(testDestinationPath, testExistingPath);
+            }
```

Comment:
Why is _this entire logic_ if-else'd and copypasted rather than the retrieval of `testExistingPath` alone?

Reply:
I did this to handle `StableContent.All` or checking a combination of different flags. So should this function only allow checking either beatmaps or others and not both at same time? Can probably use bool then instead of `StableContent`.

Reply:
Using the `StableContent` enum introduces unnecessary complexity. I would suggest changing the signature of this method to `bool CheckHardLinkAvailability(out bool canLinkBeatmaps)` and extract the underlying check logic to a local function as so:
```csharp
canLinkBeatmaps = checkAvailability(stableStorage.GetSongStorage());

return checkAvailability(stableStorage);

bool checkAvailability(Stoarge storage)
{
    string testExistingPath = storage.GetFullPath(string.Empty);
    string testDestinationPath = desktopGameHost.Storage.GetFullPath(string.Empty);

    return HardLinkHelper.CheckAvailability(testDestinationPath, testExistingPath);
}
```

Then the import from stable screen would decide what to show based on the results here (returned result shows whether data of the stable folder can be linked, while out parameter shows whether data of the beatmaps folder can be linked).

Reply:
That's missing the forest for the trees. `StableContent` is nowhere as much of an issue as trying to explain to the user all of the minute complications of them having shipped off part of their files to another drive and what that means for hard link migration.

Reply:
The enum was just one part of it focusing on the unnecessary use of the if conditionals. I've also proposed a better method of checking availability without basically duplicating code (use of local methods).